apps/tcp_tunnel_device/src/device_event_handler.c

```diff
@@ -23,15 +25,17 @@ void device_event_handler_deinit(struct device_event_handler* handler)
 
 void device_event_handler_blocking_listener(struct device_event_handler* handler)
 {
-    while(true) {
-        nabto_device_listener_device_event(handler->listener, handler->future, &handler->event);
-        NabtoDeviceError ec = nabto_device_future_wait(handler->future);
-        if (ec != NABTO_DEVICE_EC_OK) {
-            return;
-        }
-        handle_event(handler, handler->event);
-        if (handler->event == NABTO_DEVICE_EVENT_CLOSED) {
-            return;
+    if (handler->future != NULL) {
```

Comment:
Maybe we should have failed earlier and just assume the future is non null here?